cores/rp2040/CoreMutex.cpp

```diff
@@ -30,12 +30,11 @@ CoreMutex::CoreMutex(mutex_t *mutex, uint8_t option) {
     _option = option;
     if (__isFreeRTOS) {
         auto m = __get_freertos_mutex_for_ptr(mutex);
-        if (__freertos_check_if_in_isr()) {
-            __freertos_mutex_take_from_isr(m);
-        } else {
-            if (!__freertos_mutex_try_take(m)) {
+        if (__freertos_check_if_in_isr() && !__freertos_mutex_take_from_isr(m)) {
```

Comment:
Here, the take-from-isr call should pass in a new CoreMutex member variable to store the portYield return value